libraries/core-react/src/components/EdsProvider/EdsProvider.stories.tsx

```diff
@@ -0,0 +1,187 @@
+import { useRef, useState } from 'react'
+import styled from 'styled-components'
+import { Story } from '@storybook/react'
+import { menu, accessible } from '@equinor/eds-icons'
+import { useArgs } from '@storybook/client-api'
+import {
+  TopBar,
+  Menu,
+  Button,
+  Icon,
+  EdsProvider,
+  EdsProviderProps,
+} from '../..'
+
+const Wrapper = styled.div`
+  margin: 32px;
+  display: grid;
+  grid-gap: 32px;
+  grid-template-columns: repeat(4, fit-content(100%));
+`
+
+export const Default: Story<EdsProviderProps['density']> = (args) => {
+  const [_, setState] = useArgs()
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+
+  const setDensity = (density: 'comfortable' | 'compact') =>
+    setState({ density })
+
+  const openMenu = () => setIsOpen(true)
+
+  const closeMenu = () => setIsOpen(false)
+
+  const onKeyPress = (e: React.KeyboardEvent<HTMLButtonElement>) => {
+    const { key } = e
+    switch (key) {
+      case 'ArrowDown':
+        isOpen ? closeMenu() : openMenu()
+        break
+      case 'ArrowUp':
+        isOpen ? closeMenu() : openMenu()
+        break
+      case 'Escape':
+        closeMenu()
+        break
+      default:
+        break
+    }
+  }
+
+  const referenceElement = useRef<HTMLButtonElement>(null)
+
+  return (
+    <div>
+      <EdsProvider {...args}>
+        <TopBar>
+          <TopBar.Header>My Application</TopBar.Header>
+          <TopBar.Actions>
+            <Button
+              ref={referenceElement}
+              variant="ghost_icon"
+              id="menuButton"
+              aria-controls="menu-on-button"
+              aria-haspopup="true"
+              aria-expanded={isOpen}
+              onClick={() => (isOpen ? closeMenu() : openMenu())}
+              onKeyDown={onKeyPress}
+            >
+              <Icon data={accessible} title="accessible"></Icon>
+            </Button>
+            <Menu
+              id="menu-on-button"
+              open={isOpen}
+              aria-labelledby="menuButton"
+              anchorEl={referenceElement.current}
+              onClose={closeMenu}
+            >
+              <Menu.Section title="Density">
+                <Menu.Item onClick={() => setDensity('comfortable')}>
+                  Comfortable
+                </Menu.Item>
+                <Menu.Item onClick={() => setDensity('compact')}>
+                  Compact
+                </Menu.Item>
+              </Menu.Section>
+            </Menu>
+          </TopBar.Actions>
+        </TopBar>
+        <Wrapper>
+          <Button>Contained</Button>
+          <Button variant="outlined">Outlined</Button>
+          <Button variant="ghost">Ghost</Button>
+          <Button variant="ghost_icon">
+            <Icon data={menu} title="Ghost icon menu"></Icon>
+          </Button>
+        </Wrapper>
+      </EdsProvider>
+    </div>
+  )
+}
+
+const sourceCode = `const [state, setState] = useState<{
```

Comment:
My initial impression is that this is a bad bad idea in terms of maintenance and keep the preview updated, why do this? I get that this was to have a clean sourc-code example but it feels excessive and prone to errors when trying to maintane updated code examples.

Maybe just have the `density` state in the actual code and comment where you set the `setArgs` with that its a storybook thing?